### PR TITLE
[Azure Analysis Services] Change Location parameter from validate set into dynamic lookup

### PR DESCRIPTION
**Pull request by taiwu**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs [hunk 1]
```diff
@@ -41,10 +41,7 @@ public class NewAnalysisServicesServer : AnalysisServicesCmdletBase
         [Parameter(ValueFromPipelineByPropertyName = true, Position = 2, Mandatory = true,
             HelpMessage = "Azure region where the server should be created.")]
         [ValidateNotNullOrEmpty]
-        [ValidateSet("North Central US", "South Central US", "Central US", "West Europe", "North Europe", "West US",
-            "East US",
-            "East US 2", "Japan East", "Japan West", "Brazil South", "Southeast Asia", "East Asia", "Australia East",
-            "Australia Southeast", IgnoreCase = true)]
+        [LocationCompleter("Microsoft.AnalysisServices")]
```
Comment (maddieclayton) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs]: The LocationCompleter requires both the provider and resource type - I think here it would require "Microsoft.AnalysisServices/servers"

Comment (taiwu) [reply src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs]: I tested it on prod cloud and gov cloud, it seemed working(at least I can provision a server and fail test works as well). Let me try adding the resource type.

Comment (taiwu) [reply src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands/NewAzureRmAnalysisServicesServer.cs]: Both "Microsoft.AnalysisServices/servers" and "Microsoft.AnalysisServices" work, since our code is resource type specific, I will update it as "Microsoft.AnalysisServices/servers".

File: src/ResourceManager/AnalysisServices/ChangeLog.md [hunk 2]
```diff
@@ -19,6 +19,9 @@
 -->
 ## Current Release
 
+## Version 0.5.1
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/ChangeLog.md]: @taiwu we will take care of the versioning based on the public API changes made to a cmdlet. You can remove the `## Version 0.5.1` header and put the below snippet under the `## Current Release` header

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -35,5 +35,5 @@
 // by using the '*' as shown below:
 
 
-[assembly: AssemblyVersion("0.5.0")]
-[assembly: AssemblyFileVersion("0.5.0")]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Properties/AssemblyInfo.cs]: @taiwu revert the changes made in this file

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -32,6 +32,6 @@
 
 [assembly: Guid("37af97fd-d603-4d52-a7a6-0885004a57fe")]
 
-[assembly: AssemblyVersion("0.5.0")]
-[assembly: AssemblyFileVersion("0.5.0")]
+[assembly: AssemblyVersion("0.5.1")]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/Properties/AssemblyInfo.cs]: @taiwu revert the changes made in this file

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -34,5 +34,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 
-[assembly: AssemblyVersion("0.5.0")]
-[assembly: AssemblyFileVersion("0.5.0")]
```
Comment (cormacpayne) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Properties/AssemblyInfo.cs]: @taiwu revert the changes made in this file

**Comment by maddieclayton** [comment]
On demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/202/
